### ISSUE TITLE
Diagnostic for const generic enum on stable suggests requirements for nightly-only feature (PartialEq, Eq)

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/rust#95150]
Given the following code ([playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6804b0c085c6fcf4fba0c2fa9d49385e)):

```rust
enum E {
    V1,
    V2,
}

struct S<const X: E>;
```

Rust gives a diagnostic suggesting deriving `PartialEq, Eq`:

```
error: `E` is forbidden as the type of a const generic parameter
 --> src/lib.rs:6:19
  |
6 | struct S<const X: E>;
  |                   ^
  |
  = note: the only supported types are integers, `bool` and `char`

error[E0741]: `E` must be annotated with `#[derive(PartialEq, Eq)]` to be used as the type of a const parameter
 --> src/lib.rs:6:19
  |
6 | struct S<const X: E>;
  |                   ^ `E` doesn't derive both `PartialEq` and `Eq`

For more information about this error, try `rustc --explain E0741`.
```

However, adding `#[derive(PartialEq, Eq)]` leads the compiler to still give an error about not supporting this.

A non-nightly compiler should not give the second error at all; it should only give the first, saying that the only supported types are integers, `bool` and `char`.

**Comment by lcnr** [comment]
To fix this issue, you can do the following steps:
- grep for "forbidden as the type of a const generic parameter"
- look at that function for a bit
- change it so that the second error isn't emitted on stable (probably actually change the control flow to start by branching on `tcx.features().adt_const_params`

In case you are interested but need more help to fix this, feel free to reach out to me on [zulip](https://rust-lang.zulipchat.com/#narrow/stream/260443-project-const-generics)
